utils/orchestrator/ecsutil.py

```diff
@@ -3,62 +3,82 @@
 # Licensed under Simplified BSD License (see LICENSE)
 
 # stdlib
-import logging
+import re
+
+# 3rd
 import requests
-import socket
 
 # project
-from utils.dockerutil import DockerUtil
-from utils.platform import Platform
-from utils.singleton import Singleton
+from .baseutil import BaseUtil
 
+ECS_AGENT_DEFAULT_PORT = 51678
+ECS_AGENT_CONTAINER_NAME = 'ecs-agent'
+ECS_AGENT_METADATA_PATH = '/v1/metadata'
+ECS_AGENT_TASKS_PATH = '/v1/tasks'
 
-log = logging.getLogger(__name__)
+AGENT_VERSION_EXP = re.compile(r'v([0-9.]+)')
 
-ECS_INTROSPECT_DEFAULT_PORT = 51678
-ECS_AGENT_CONTAINER_NAME = 'ecs-agent'
 
-class ECSUtil:
-    __metaclass__ = Singleton
+def ECS_AGENT_VALIDATION(r):
```

Comment:
maybe make it a static method instead? It's strange to have it here and all caps.